pkg/asset/manifests/aws/cluster.go

```diff
@@ -151,19 +152,53 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 				PresignedURLDuration: &metav1.Duration{Duration: 1 * time.Hour},
 			},
 			ControlPlaneLoadBalancer: &capa.AWSLoadBalancerSpec{
-				Name:             ptr.To(clusterID.InfraID + "-ext"),
+				Name:             ptr.To(clusterID.InfraID + "-int"),
 				LoadBalancerType: capa.LoadBalancerTypeNLB,
-				Scheme:           &capa.ELBSchemeInternetFacing,
+				Scheme:           &capa.ELBSchemeInternal,
 				AdditionalListeners: []capa.AdditionalListenerSpec{
 					{
 						Port:     22623,
 						Protocol: capa.ELBProtocolTCP,
 					},
 				},
+				IngressRules: []capa.IngressRule{
+					{
+						Description: "Machine Config Server internal traffic from cluster",
+						Protocol:    capa.SecurityGroupProtocolTCP,
+						FromPort:    22623,
+						ToPort:      22623,
+						CidrBlocks:  []string{mainCIDR.String()},
+					},
+					{
+						Description: "Kubernetes API Server traffic for internal acccess",
+						Protocol:    capa.SecurityGroupProtocolTCP,
+						FromPort:    6443,
+						ToPort:      6443,
+						CidrBlocks:  []string{mainCIDR.String()},
+					},
+				},
 			},
 		},
 	}
 
+	if installConfig.Config.Publish == types.ExternalPublishingStrategy {
```

Comment:
@r4f4 here is the changes I needed to apply considering the CAPA limitations* we are discussing on Slack.

```suggestion
				IngressRules: []capa.IngressRule{
					{
						Description: "Machine Config Server internal traffic from cluster",
						Protocol:    capa.SecurityGroupProtocolTCP,
						FromPort:    22623,
						ToPort:      22623,
						CidrBlocks:  []string{mainCIDR.String()},
					},
				},
			},
		},
	}

	if installConfig.Config.Publish == types.ExternalPublishingStrategy {
	    awsCluster.Spec.ControlPlaneLoadBalancer.IngressRules = append(awsCluster.Spec.ControlPlaneLoadBalancer.IngressRules, capa.IngressRule{
			Description: "Kubernetes API Server traffic for public access",
			Protocol:    capa.SecurityGroupProtocolTCP,
			FromPort:    6443,
			ToPort:      6443,
			CidrBlocks:  []string{"0.0.0.0/0"},
		})
```

*Limitations we found so far:
- When defining the `ControlPlaneLoadBalancer.IngressRules` for API w/ internal access, the CAPA will complain due duplicated rule. CAPA Automatically add the ingress rule <VPC CIDR> 6443/TCP
- When defining the `SecondaryControlPlaneLoadBalancer.IngressRules` the CAPA is ignoring